File: KitchenRP.DataAccess/Models/Reservation.cs
using System.Collections;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using NodaTime;

namespace KitchenRP.DataAccess.Models
{
    public class Reservation
    {
        public long? Id { get; set; }
        public Instant StartTime { get; set; }
        public Instant EndTime { get; set; }
        public User Owner { get; set; }
        public Resource ReservedResource { get; set; }
        public bool AllowNotifications { get; set; }

        public ICollection<StatusChange> StatusChanges { get; set; }
    }

    internal class ReservationTypeConfiguration : IEntityTypeConfiguration<Reservation>
    {
        public void Configure(EntityTypeBuilder<Reservation> builder)
        {
            builder
                .HasOne(r => r.Owner)
                .WithMany()
                .HasForeignKey("OwnerId");

            builder
                .HasOne(r => r.ReservedResource)
                .WithMany()
                .HasForeignKey("ReservedResourceId");
        }
    }
}

File: kitchenRP-frontend/kitchenRP/src/app/types/reservation.ts
import {User} from "./user";
import {Resource} from "./resource";
import {Status} from "./status";

export interface Reservation{
    id: number;
    //date: any;
    //duration: any;
    startTime: string;
    endTime: string;
    userId: number;
    resourceId: number;
    status: Status;
    statuses: string;
    owner: User;
    reservedResource: Resource;
}

export class NewReservation{
    startTime: string;
    endTime: string;
    userId: number;
    resourceId: number;
    allowNotifications: boolean;
}
